src/c++/perf_analyzer/perf_analyzer.cc

```diff
@@ -52,6 +52,44 @@ SignalHandler(int signum)
     exit(0);
   }
 }
+
+bool
+IsLLMModel(
+    const std::shared_ptr<pa::ModelParser>& parser,
+    const pa::PAParamsPTR& params)
+{
+  bool is_llm_from_user = params->is_llm if (is_llm_from_user)
+  {
+    return true;
+  }
+
+  bool is_llm = false;
+  // check if its decoupled
+  is_llm =
+      is_llm || (parser->IsDecoupled() && !params->profile_export_file.empty());
```

Comment:
Will the greater changes here have any negative implications on the outputs for non-decoupled LLM models? ("offline" scenario)

Reply:
I think this will require testing of some sort to ensure appropriate behavior.